### PR TITLE
Issue #4365: Split and Organize Checkstyle inputs by Test for checks …

### PR DESCRIPTION
**Pull request by sharang108**
Related to Issue #4365
 
> subtask of #3891
> 
> move / do a copy of all test input files to a separate folder for all checks in header package.
> Not any other Check should use new

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/InputRegexpSmallHeader.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.puppycrawl.tools.checkstyle.checks.header.regexpheader;
+
+
+/**
+ */
+public class InputRegexpSmallHeader {}
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/InputRegexpSmallHeader.java]: Inputs must be named after their tests. Test is `RegexpHeaderCheckTest` so inputs' name must start with `InputRegexpHeader`.
Please also keep class name as file name.

Please also do the same with the `regexp.header` files.

Comment (sharang108) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/InputRegexpSmallHeader.java]: So I should name 'InputRegexpSmallHeader' to 'InputRegexpHeader'? that would create a conflict there is already a class named InputRegexpHeader.

Comment (Vladlis) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/header/regexpheader/InputRegexpSmallHeader.java]: @sharang108
It must **start** with `InputRegexpHeader`, so you can append anything appropriate.

**Comment by rnveach** [comment]
@sharang108 Can we also rename the header files?
Example: `regexp.header`

**Comment by sharang108** [comment]
@rnveach  I don't think there is a unique short name which we can associate with each and every file in the package like `InputRegexpHeaderSmallHeader`.

**Comment by rnveach** [comment]
@sharang108 I am not picky about whats in the middle, just make as best as you can.
You can choose something based on the test's name or just use a numbering scheme.
I am only referring to the files with the extension `header`.

**Comment by rnveach** [comment]
@sharang108 Please notify us when you have made changes.
We don't get updates when commits are changed/added.

**Comment by rnveach** [comment]
@sharang108 @romani Nothing has changed that I can see.
I still see code like `getPath("regexp.header")`. I expect to see something like `getPath("InputRegexpHeader.header")` so that the file is named after the test.
All inputs should follow test name, not just java files.

**Comment by sharang108** [comment]
@romani @rnveach  I've made changes as you asked

**Comment by romani** [comment]
@sharang108 , please squash all in one commit.
In PR should be only one commit.

**Comment by sharang108** [comment]
@romani I've removed the multiple Commits @rnveach

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java [hunk 2]
```diff
@@ -204,123 +204,123 @@ public void testFailureForMultilineRegexp() throws Exception {
...
         final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
-        verify(checkConfig, getPath("InputRegexpHeader8.java"), expected);
+        verify(checkConfig, getPath("InputRegexpHeaderIgnoreLinesSorted.java"), expected);
     }
 
     @Test
     public void testHeaderWithInvalidRegexp() throws Exception {
         final DefaultConfiguration checkConfig = createCheckConfig(RegexpHeaderCheck.class);
-        checkConfig.addAttribute("headerFile", getPath("regexp.invalid.header"));
+        checkConfig.addAttribute("headerFile", getPath("InputRegexp.invalid.header"));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/header/RegexpHeaderCheckTest.java]: Missed the full file name.
`InputRegexp.invalid.header` => `InputRegexpHeader.invalid.header`

**Comment by sharang108** [comment]
@rnveach Done! you can now approve this PR @romani

**Pull request merged**